awsclilinter/awsclilinter/rules/pagination_rule.py

```diff
@@ -0,0 +1,50 @@
+from typing import List
+
+from ast_grep_py.ast_grep_py import SgRoot
+
+from awsclilinter.rules import LintFinding, LintRule
+
+
+class PaginationRule(LintRule):
+    """Detects AWS CLI commands missing --no-cli-paginate flag."""
+
+    @property
+    def name(self) -> str:
+        return "no-cli-paginate"
+
+    @property
+    def description(self) -> str:
+        return (
+            "AWS CLI v2 uses pagination by default for commands that return large result sets. "
+            "Add --no-cli-paginate to disable pagination and match v1 behavior."
```

Comment:
Users can also disable pagination via env variable or config file option, which they may be already doing. Would that make detection a false positive?. If so, how do we account for that?